integration/management-api-tests/src/test/java/org/wso2/micro/integrator/api/ConfigsResourceTestCase.java

```diff
@@ -0,0 +1,290 @@
+
+        super.init();
+        offset = portOffset + 10;
+    }
+
+    /**
+     * Test retrieve correlation configs. This will check whether the management API retrieves
+     * proper correlation config status which is 'false' and reads the correlation log file
+     * which should be blank.
+     *
+     * @throws IOException                  the io exception
+     * @throws AutomationFrameworkException the automation framework exception
+     */
+    @Test(groups = {"wso2.esb"}, description = "Test get configs resource for correlation")
+    public void testRetrieveCorrelationConfigs() throws IOException, AutomationFrameworkException {
+
+        startNewServer(false);
+        sendGetRequest(offset, false);
+        String apiLogFilePath = carbonHome + File.separator + "repository"
+                + File.separator + "logs" + File.separator + "correlation.log";
+        BufferedReader bufferedReader = new BufferedReader(new FileReader(apiLogFilePath));
+        String logLine = bufferedReader.readLine();
+        Assert.assertNull(logLine);
+    }
+
+    /**
+     * Test update correlation configs. This will enable the correlation configs using the
+     * management API. It will check whether the update was successful by retrieving the
+     * correlation logs status from the management API and reading the correlation log file
+     * for logs.
+     *
+     * @throws IOException the io exception
+     */
+    @Test(groups = {"wso2.esb"}, dependsOnMethods = {"testRetrieveCorrelationConfigs"},
+            description = "Test get configs resource for correlation")
+    public void testUpdateCorrelationConfigsEnable() throws IOException {
+
+        sendPutRequest(offset, true);
+        sendGetRequest(offset, true);
+
+        String apiLogFilePath = carbonHome + File.separator + "repository"
+                + File.separator + "logs" + File.separator + "correlation.log";
+        BufferedReader bufferedReader = new BufferedReader(new FileReader(apiLogFilePath));
+        String logLine = bufferedReader.readLine();
+        Assert.assertNotNull(logLine);
+        while ((logLine = bufferedReader.readLine()) != null) {
+            Assert.assertTrue(logLine.contains("HTTP State Transition") ||
+                    logLine.contains("ROUND-TRIP LATENCY") || logLine.contains("Thread switch latency"));
+        }
+    }
+
+    /**
+     * Test update correlation configs disable. This will disable the correlation logs using the
+     * Management API. It will check whether the update was successful by retrieving the correlation
+     * logs status from the management API and checking the correlation log file for no logs.
+     *
+     * @throws IOException the io exception
+     */
+    @Test(groups = {"wso2.esb"}, dependsOnMethods = {"testUpdateCorrelationConfigsEnable"},
+            description = "Test get configs resource for correlation")
+    public void testUpdateCorrelationConfigsDisable() throws IOException {
```

Comment:
Shall we improve this test case to make sure correlation logs are disabled at the runtime as well? Maybe by checking the timestamp of the `correlation.log` file